bin/node/src/service.rs

```diff
@@ -64,6 +64,21 @@ impl<N: BaseArithmetic> BackoffAuthoringBlocksStrategy<N> for LimitNonfinalized
     }
 }
 
+fn revert_unfinalized_blocks(client: &FullClient) {
+    let last_finalized = client.info().finalized_number;
+    let best_block = client.info().best_number;
+    // 20 so there are no problems with finality session
+    const REVERT_THRESHOLD: u32 = 20;
+
+    let blocks_count_to_revert = best_block.saturating_sub(last_finalized + REVERT_THRESHOLD);
+
+    debug!("Reverting {} not finalized blocks", blocks_count_to_revert);
+    let new_best = client
```

Comment:
I think it returns the number of blocks that were reverted, not the new best block number.

Reply:
good catch, fixed